### PR TITLE
Clean up messaging/RPC port configuration and docs

### PR DESCRIPTION
**Pull request by rnicoll**
This makes the messaging port configuration consistent between code and configuration to avoid using different terminology, adds an "rpcPort" option to build.gradle and configures it everywhere so port numbers are explicit instead of implicit.

It may be easier to change `messagingAddress` in code to `artemisAddress`, but I think it's worth making the two Artemis addresses (messaging and RPC) clearly labelled, rather than trying to minimise change at this point.

### CODE REVIEW

File: docs/source/creating-a-cordapp.rst [hunk 1]
```diff
@@ -214,24 +214,27 @@ is a three node example;
             name "Controller"
             nearestCity "London"
             advertisedServices = [ "corda.notary.validating" ]
-            artemisPort 12345
-            webPort 12346
+            messagingPort 12345
```
Comment (Clintonio) [inline docs/source/creating-a-cordapp.rst]: Can we switch these ports to be the `100xy` rang. It's annoyingly arbitrary right now.

File: samples/raft-notary-demo/build.gradle [hunk 2]
```diff
@@ -89,7 +89,7 @@ task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['build', 'generat
         name "Party"
         nearestCity "London"
         advertisedServices = []
-        artemisPort 10002
+        messagingPort 10002
```
Comment (Clintonio) [inline samples/raft-notary-demo/build.gradle]: These nodes have no RPC address. How will the RPC based webserver connect? Is it still valid to connect to the messaging port?

Comment (mikehearn) [inline samples/raft-notary-demo/build.gradle]: It's not supposed to be.

File: docs/source/corda-configuration-file.rst [hunk 3]
```diff
@@ -74,14 +75,17 @@ path to the node's base directory.
...
-:artemisAddress: The host and port on which the node is available for protocol operations over ArtemisMQ.
+:messagingAddress: The host and port on which the node is available for protocol operations over ArtemisMQ.
 
     .. note:: In practice the ArtemisMQ messaging services bind to all local addresses on the specified port. However,
         note that the host is the included as the advertised entry in the NetworkMapService. As a result the value listed
         here must be externally accessible when running nodes across a cluster of machines.
 
 :messagingServerAddress: The address of the ArtemisMQ broker instance. If not provided the node will run one locally.
 
+:rpcAddress: The address of the Artemis RPC broker instance. If not provided the node will run one locally, one port higher than
```
Comment (mikehearn) [inline docs/source/corda-configuration-file.rst]: This description doesn't sound right: it's not the address of the broker, it's the port on which RPCs are received, surely? Indeed, "messagingAddress" and "rpcAddress" both seem wrong: they should just be port numbers?

**Comment by mikehearn** [comment]
Is it really needed for people to install the gradle plugins themselves? Surely they can just bump their dependency version?

**Comment by rnicoll** [comment]
Confirming the upgrade instructions at the moment. It might be for the release you don't, but will have to manually run the install until artifacts are in Maven.

**Comment by rnicoll** [comment]
Talked to Clinton about the process - once we're happy with this, I'll release the new version of the plugins to Bintray so they're ready before we merge the PR.

**Comment by Clintonio** [comment]
WRT to the gradle plugins the cycle I've been doing since open sourcing has been to just publish them to bintray, bump the version in publish.properties (shared by both Corda and the plugins), the template and tutorial. Nobody should ever have to run the gradle plugins installation except people developing them (which ideally wouldn't be too common either).

File: docs/source/corda-configuration-file.rst (hunk 3 above)
Comment (patrickkuo) [inline docs/source/corda-configuration-file.rst]: I would like change messagingAddress, rpcAddress and webAdress to hostName, messagingPort, rpcPort and webPort, but these changes will touch lot of classes, i think its better to do this in a separate PR in cleanup week?

**Comment by patrickkuo** [comment]
Thanks @rnicoll for picking this up while I am away! 
I have tested and fixed all demos, with both driver DSL and Cordform, and also addressed all issues.
This PR is ready for more reviews.

File: config/dev/generalnodea.conf [hunk 4]
```diff
@@ -2,7 +2,7 @@ myLegalName : "Bank A"
 nearestCity : "London"
 keyStorePassword : "cordacadevpass"
 trustStorePassword : "trustpass"
-artemisAddress : "localhost:31337"
```
Comment (rnicoll) [inline config/dev/generalnodea.conf]: Can we renumber these to match port number ranges used elsewhere please?

File: node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt [hunk 5]
```diff
@@ -152,26 +148,26 @@ abstract class MQSecurityTest : NodeBasedTest() {
...
-        val client = SimpleMQClient(target, config)
+    fun clientTo(target: HostAndPort, sslConfiguration: SSLConfiguration? = configureTestSSL()): SimpleMQClient {
+        val client = SimpleMQClient(target, sslConfiguration)
         clients += client
         return client
     }
 
     fun loginToRPC(target: HostAndPort, rpcUser: User): SimpleMQClient {
-        val client = clientTo(target)
+        val client = clientTo(target, null)
```
Comment (rnicoll) [inline node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt]: Do we never use SSL with RPC? I'd be inclined to take the `SSLConfiguration` in as a parameter to the `loginToRPC` function

File: node/src/main/kotlin/net/corda/node/internal/Node.kt [hunk 6]
```diff
@@ -24,6 +24,7 @@ import net.corda.node.services.messaging.ArtemisMessagingServer
 import net.corda.node.services.messaging.NodeMessagingClient
 import net.corda.node.services.transactions.*
 import net.corda.node.utilities.AddressUtils
+import net.corda.node.services.transactions.*
```
Comment (rnicoll) [inline node/src/main/kotlin/net/corda/node/internal/Node.kt]: Is this used?

File: node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingComponent.kt [hunk 7]
```diff
@@ -151,7 +152,7 @@ abstract class ArtemisMessagingComponent() : SingletonSerializeAsToken() {
                 TransportConstants.PROTOCOLS_PROP_NAME to "CORE,AMQP"
         )
 
-        if (config != null) {
+        if (config != null && enableSSL) {
```
Comment (rnicoll) [inline node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingComponent.kt]: Should perhaps throw an exception if told to enable SSL but can't generate a configuration?

File: node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt (hunk 5 above)
Comment (patrickkuo) [inline node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt]: We use SSL with RPC when we login as NODE_USER(used by the web server), where the login module checks the certificate to verify its from the same machine.
The ssl config object is a member variable in `ArtemisMessangingComponent` moving it as a parameter to the method will require lot of changes...

Comment (rnicoll) [inline node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt]: Can't we make the function signature:

`fun loginToRPC(target: HostAndPort, rpcUser: User, config: SSLConfiguration? = null): SimpleMQClient`

and then it should work as-is?

Comment (patrickkuo) [inline node/src/integration-test/kotlin/net/corda/services/messaging/MQSecurityTest.kt]: Sure, will change this. I misread the comment and thought we were talking about changes in `CordaRPCClient`...

**Comment by shamsasari** [comment]
`messagingAddress` is too vague. Can we change it to `peerAddress` or `p2pAddress`? What does everyone else think?

**Comment by rnicoll** [comment]
I'd say `p2pAddress` if we're going to rename, `peerAddress` to me is the address of your peer

**Comment by mikehearn** [comment]
myNetworkAddress?

**Comment by rnicoll** [comment]
We have multiple network addresses dependent on use case (hence messaging/p2p, RPC and web). I know people don't like how p2pAddress sounds, but it seems like the most accurate fit.

**Comment by patrickkuo** [comment]
I have addressed all issues.
@Clintonio branches for template and tutorial has been created, i am waiting for the web server fix PR in both repo.

https://github.com/corda/cordapp-tutorial/tree/pat-rpc-port-changes
https://github.com/corda/cordapp-template/tree/pat-rpc-port-changes

File: docs/source/corda-configuration-file.rst [hunk 8]
```diff
@@ -74,13 +75,15 @@ path to the node's base directory.
...
+:messagingServerAddress: The address of the ArtemisMQ broker instance. If not provided the node will run one locally.
+
+:p2pAddress: The host and port on which the node is available for protocol operations over ArtemisMQ.
 
     .. note:: In practice the ArtemisMQ messaging services bind to all local addresses on the specified port. However,
         note that the host is the included as the advertised entry in the NetworkMapService. As a result the value listed
         here must be externally accessible when running nodes across a cluster of machines.
 
-:messagingServerAddress: The address of the ArtemisMQ broker instance. If not provided the node will run one locally.
+:rpcAddress: The address of the Artemis RPC broker instance. If not provided the node will start without RPC connectivity.
```
Comment (shamsasari) [inline docs/source/corda-configuration-file.rst]: The RPC is not running a broker, which is what this implies. A better description might be `The address of the RPC system on which RPC requests can be made to the node. If not provided then the node will run without RPC.`

File: docs/source/release-notes.rst [hunk 9]
```diff
@@ -4,6 +4,15 @@ Release notes
 Here are release notes for each snapshot release from M9 onwards. This includes guidance on how to upgrade code from
 the previous milestone release.
 
+Milestone 10
+------------
+
+Important: There are configuration changes in M10 due to the split of the Artemis port into separate messaging and RPC
```
Comment (shamsasari) [inline docs/source/release-notes.rst]: ... into separate P2P and RPC ...

File: docs/source/setting-up-a-corda-network.rst [hunk 10]
```diff
@@ -42,11 +42,13 @@ An example configuration:
 
 The most important fields regarding network configuration are:
 
-* ``artemisAddress``: This specifies a host and port. Note that the address bound will **NOT** be ``my-corda-node``,
-  but rather ``::`` (all addresses on all interfaces). The hostname specified is the hostname *that must be externally
-  resolvable by other nodes in the network*. In the above configuration this is the resolvable name of a machine in a vpn.
-* ``webAddress``: The address the webserver should bind. Note that the port should be distinct from that of ``artemisAddress``
-                  if they are on the same machine.
+* ``p2pAddress``: This specifies a host and port to which Artemis will bind for messaging with other nodes. Note that the
+  address bound will **NOT** be ``my-corda-node``, but rather ``::`` (all addresses on all interfaces). The hostname specified
```
Comment (shamsasari) [inline docs/source/setting-up-a-corda-network.rst]: (i.e. all network interfaces)

File: gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy [hunk 11]
```diff
@@ -80,13 +80,23 @@ class Node {
     }
 
     /**
-     * Set the artemis port for this node.
+     * Set the Artemis messaging port for this node.
```
Comment (shamsasari) [inline gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy]: messaging -> P2P

File: gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy [hunk 12]
```diff
@@ -143,7 +153,7 @@ class Node {
      * @return This node's artemis address.
      */
     String getArtemisAddress() {
```
Comment (shamsasari) [inline gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy]: getP2PAddress()

File: gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy [hunk 13]
```diff
@@ -143,7 +153,7 @@ class Node {
      * @return This node's artemis address.
```
Comment (shamsasari) [inline gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy]: P2P

File: node/src/main/kotlin/net/corda/node/Corda.kt [hunk 14]
```diff
@@ -138,6 +141,15 @@ fun main(args: Array<String>) {
     exitProcess(0)
 }
 
+private fun checkConfigVersion(conf: Config) {
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/Corda.kt]: Add a TODO to remove this check some milestones down the road

**Comment by Clintonio** [comment]
@patrickkuo Those PRs in the dependent repos are merged.

**Comment by patrickkuo** [comment]
thanks @shamsasari @Clintonio, PR issues addressed and PR created for tutorial and template.
https://github.com/corda/cordapp-template/pull/58
https://github.com/corda/cordapp-tutorial/pull/28
